README.md

```diff
@@ -1,14 +1,75 @@
+This library is a port of the [Microsoft Azure IoT device SDK for C](https://github.com/Azure/azure-iot-sdks/blob/master/c/readme.md) to Arduino. It allows you to use several Arduino compatible boards with Azure IoT Hub.
 
-For the Arduino/Genuino [MKR1000](https://www.arduino.cc/en/Main/ArduinoMKR1000) or [Zero](https://www.arduino.cc/en/Main/ArduinoBoardZero) and [WiFi Shield 101](https://www.arduino.cc/en/Main/ArduinoWiFiShield101) only.
+Currently supported hardware:
+- Atmel SAMD Based boards
+  - Arduino/Genuino [MKR1000](https://www.arduino.cc/en/Main/ArduinoMKR1000) (pending release)
+  - Adafruit [Feather M0](https://www.adafruit.com/products/3010)
+  - [Zero](https://www.arduino.cc/en/Main/ArduinoBoardZero) and [WiFi Shield 101](https://www.arduino.cc/en/Main/ArduinoWiFiShield101) 
+- ESP8266 based boards with [esp8266/arduino](https://github.com/esp8266/arduino)
+  - SparkFun [Thing](https://www.sparkfun.com/products/13711)
+  - Adafruit [Feather Huzzah](https://www.adafruit.com/products/2821)
 
 [![Microsoft Azure Certified][Microsoft-Azure-Certified-Badge]][azure-certifiedforiot]
 
-## Getting started
+## Prerequisites
 
-See the [Microsoft Azure IoT device SDK for C - Arduino WiFi Shield 101 and MKR1000 Setup Guide](https://github.com/Azure/azure-iot-sdks/blob/master/c/doc/run_sample_on_arduino_wifi101.md).
+You should have the following ready before beginning with any board:
+-   [Setup your IoT hub](https://github.com/Azure/azure-iot-sdks/blob/master/doc/setup_iothub.md)
+-   [Provision your device and get its credentials](https://github.com/Azure/azure-iot-sdks/blob/master/doc/manage_iot_hub.md)
+-   [Arduino IDE 1.6.8](https://www.arduino.cc/en/Main/Software) (currently this is available only as the hourly build)
+-  Install the AzureIoT library
+    * Currently this must be installed from source control by cloning the repo into your arduino libraries folder.  An example is below:
+
+        ```
+        cd C:\Program Files (x86)\Arduino\libraries
+        git clone <the URI for this repository>
+        ```
+Then reference the section appropriate for your hardware below.
+
+## ESP8266
+##### Sparkfun Thing, Adafruit Feather Huzzah, or generic ESP8266 board
+
+1. Install esp8266 board support into your Arduino IDE.
+    * Start Arduino and open Preferences window.
+    * Enter `http://arduino.esp8266.com/stable/package_esp8266com_index.json` into Additional Board Manager URLs field. You can add multiple URLs, separating them with commas.
+    * Open Boards Manager from Tools > Board menu and install esp8266 platform 2.1.0 or later 
+    * Select your ESP8266 board from Tools > Board menu after installation
+    
+2. Copy and reference the sample provided in library directory 
+    ```
+    \examples\esp8266\simplesample_http\simplesample_http.ino
```

Comment:
What do you think about changing this for use the Arduino IDE examples menu? ```File -> Examples -> AzureIoT``` (same applies to other boards below)